txtorcon/torcontrolprotocol.py

```diff
@@ -716,13 +716,13 @@ def _bootstrap(self, *args):
         callback.
         """
 
-        # unfortunately I don't see a way to get this from the runing
-        # tor like the events...so this was taken from some version
-        # of the control-spec and must be kept up-to-date (or accpet
-        # any signal name and just wait for the reply?
-        # FIXME XXX there is now "GETINFO signal/names"
-        self.valid_signals = ["RELOAD", "DUMP", "DEBUG", "NEWNYM",
-                              "CLEARDNSCACHE"]
+        try:
+            self.valid_signals = yield self.get_info('signal/names')
+            self.valid_signals = self.valid_signals['signal/names']
+        except TorProtocolError:
+            self.valid_signals = ["RELOAD", "DUMP", "DEBUG", "NEWNYM",
+                                  "CLEARDNSCACHE"]
+            raise RuntimeError("Tor version does not support signal/names")
 
```

Comment:
This isn't an error; we've figured out the valid_signals list (i.e. using the static list, not the one from getinfo).